extensions/TheShovel/ShovelUtils.js

```diff
@@ -216,15 +216,30 @@
     }
 
     getlist({ TEXT }) {
-      return JSON.stringify(vm.runtime.getTargetForStage().lookupVariableByNameAndType(TEXT, 'list').value);
+      try{return JSON.stringify(vm.runtime.getTargetForStage().lookupVariableByNameAndType(TEXT, 'list').value);}
+      catch(error){
+        return ""
+      }
     }
     setlist({ TEXT, NAME }) {
-      const temp = JSON.parse(TEXT);
-      vm.runtime.getTargetForStage().lookupVariableByNameAndType(NAME, 'list').value = temp;
+      try{const temp = JSON.parse(TEXT);
+        if (Array.isArray(JSON.parse(TEXT))){
+      vm.runtime.getTargetForStage().lookupVariableByNameAndType(NAME, 'list').value = temp;}
+    else{return ""}}
+      catch(error){
+        return ""
+      }
     }
 
     setedtarget({ NAME }) {
-      vm.setEditingTarget(vm.runtime.getSpriteTargetByName(NAME).id);
+      //I know this might cause sprites called "stage" to be ignored. But lets be real, who names their sprite "stage"?
+      try{if (NAME.toLowerCase() === "stage"){ 
+        vm.setEditingTarget(vm.runtime.getTargetForStage().id);}
+      else
+      {vm.setEditingTarget(vm.runtime.getSpriteTargetByName(NAME).id);}}
+      catch(error){
+        return ""
+      }
```

Comment:
```suggestion
      let target;

      //I know this might cause sprites called "stage" to be ignored. But lets be real, who names their sprite "stage"?
      if (NAME.toLowerCase() === "stage") { 
        target = vm.runtime.getTargetForStage();
      } else {
        target = vm.runtime.getSpriteTargetByName(NAME);
      }
      if (target) {
        vm.setEditingTarget(target.id);
      }
```